### PR TITLE
KAFKA-9308: Reworded the ssl part of the security documentation 

### PR DESCRIPTION
**Pull request by soenkeliebau**
This is to fix various issues (mainly as noted by this jira, the problem that SAN extension values are not copied to certificates) and add some recommendations.

Build the page and reviewed it, used Intellij HTML syntax checker to ensure valid HTML syntax.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by soenkeliebau** [comment]
@gwenshap & @guozhangwang I noticed that the two of you have written and reviewed the current state of this documentation, so I figured you might be best suited to review this PR.

**Comment by soenkeliebau** [comment]
bump @gwenshap & @guozhangwang

File: docs/security.html [hunk 1]
```diff
@@ -37,114 +37,291 @@ <h3><a id="security_overview" href="#security_overview">7.1 Security Overview</a
...
-            <pre class="brush: bash;">
-            openssl req -new -x509 -keyout ca-key -out ca-cert -days 365</pre>
+            After this step each machine in the cluster has a public/private key pair which can already be used to encrypt traffic and a certificate
+            signing request, which is the basis for creating a certificate. To add authentication capabilities this signing request needs to be signed
+            by a trusted authority, which will be created in this step.
+
+            <p>A certificate authority (CA) is responsible for signing certificates. CAs works likes a government that issues passports—the government
```
Comment (mimaison) [inline docs/security.html]: Are we missing spaces around `-`?

File: docs/security.html [hunk 2]
```diff
@@ -37,114 +37,291 @@ <h3><a id="security_overview" href="#security_overview">7.1 Security Overview</a
...
+            stamps (signs) each passport so that the passport becomes difficult to forge. Other governments verify the stamps to ensure the passport is
+            authentic. Similarly, the CA signs the certificates, and the cryptography guarantees that a signed certificate is computationally difficult to
+            forge. Thus, as long as the CA is a genuine and trusted authority, the clients have a strong assurance that they are connecting to the authentic
+            machines.
+
+            <p>For this guide we will be our own Certificate Authority, when setting up a production cluster in a corporate environment these certificates would
```
Comment (mimaison) [inline docs/security.html]: I feel like breaking this sentence in two would make it easier to read.

File: docs/security.html [hunk 3]
```diff
@@ -37,114 +37,291 @@ <h3><a id="security_overview" href="#security_overview">7.1 Security Overview</a
...
+            usually be signed by a corporate CA that is trusted throughout the company, please see <a href="#security_ssl_production">Common Pitfalls in
+            Production</a> for some things to consider for this case.
+
+            <p>Due to a <a href="https://www.openssl.org/docs/man1.1.1/man1/x509.html#BUGS">bug</a> in OpenSSL the x509 module will not copy requested
```
Comment (mimaison) [inline docs/security.html]: Can we add a comma after OpenSSL?

File: docs/security.html [hunk 4]
```diff
@@ -37,114 +37,291 @@ <h3><a id="security_overview" href="#security_overview">7.1 Security Overview</a
...
 
-        <li><h4><a id="security_ssl_signing" href="#security_ssl_signing">Signing the certificate</a></h4>
-            The next step is to sign all certificates generated by step 1 with the CA generated in step 2. First, you need to export the certificate from the keystore:
+            The next step is to add the generated CA to the **clients' truststore** so that the clients can trust this CA:
+            <pre class="brush: bash;">
+                keytool -keystore client.truststore.jks -alias CARoot -import -file ca-cert
+            </pre>
+
+            <b>Note:</b>
+            If you configure the Kafka brokers to require client authentication by setting ssl.client.auth to be "requested" or "required" on the
```
Comment (mimaison) [inline docs/security.html]: I think `in the` instead of `on the` is more correct

**Review by mimaison — CHANGES_REQUESTED** [review]
Thanks @soenkeliebau ! It looks great overall.
I've left a few minor comments

File: docs/security.html (hunk 4 above)
Comment (soenkeliebau) [reply docs/security.html]: I agree, fixed!

File: docs/security.html (hunk 1 above)
Comment (soenkeliebau) [reply docs/security.html]: Jep, fixed.

File: docs/security.html (hunk 3 above)
Comment (soenkeliebau) [reply docs/security.html]: done

File: docs/security.html (hunk 2 above)
Comment (soenkeliebau) [reply docs/security.html]: I've tried splitting it to make it more readable.

**Comment by soenkeliebau** [comment]
I've addressed the comments.

**Review by mimaison — APPROVED** [review]

**Pull request merged**